### PR TITLE
Add deprecated packs to IGNORED_FILES list

### PR DESCRIPTION
**Pull request by darkushin**
Add all hidden packs to the list of IGNORED_FILES when installing all packs in a nightly build.

### CODE REVIEW

File: Tests/Marketplace/search_and_install_packs.py [hunk 1]
```diff
@@ -449,6 +449,15 @@ def install_all_content_packs_for_nightly(client: demisto_client, host: str, ser
...
 
+    # Add deprecated packs to IGNORED_FILES list:
+    for pack_id in os.listdir(PACKS_FULL_PATH):
+        metadata_path = os.path.join(PACKS_FULL_PATH, pack_id, PACK_METADATA_FILE)
+        with open(metadata_path, 'r') as json_file:
+            pack_metadata = json.load(json_file)
+            hidden = pack_metadata.get('hidden', False)
+            if hidden:
+                IGNORED_FILES.append(pack_id)
+
```
Comment (hod-alpert) [inline Tests/Marketplace/search_and_install_packs.py:460]: Let's make it a method called `is_pack_hidden`
and add unittests for that method.

File: Tests/Marketplace/search_and_install_packs.py [hunk 2]
```diff
@@ -39,6 +39,21 @@ def get_pack_display_name(pack_id: str) -> str:
...
 
+def is_pack_hidden(pack_id: str) -> bool:
+    """
+    Check if the given pack is deprecated.
+
+    :param pack_id: ID of the pack.
+    :return: True if the pack is deprecated, i.e. has 'hidden: true' field, False otherwise.
+    """
+    metadata_path = os.path.join(PACKS_FULL_PATH, pack_id, PACK_METADATA_FILE)
+    if pack_id and os.path.isfile(metadata_path):
```
Comment (yuvalbenshalom) [inline Tests/Marketplace/search_and_install_packs.py:50]: if you didn't even find the pack metadata, this needs to be logged as a warning

**Review by yuvalbenshalom — CHANGES_REQUESTED** [review]
see my comment

File: Tests/Marketplace/search_and_install_packs.py [hunk 3]
```diff
@@ -449,6 +466,11 @@ def install_all_content_packs_for_nightly(client: demisto_client, host: str, ser
     production_bucket = storage_client.bucket(GCPConfig.PRODUCTION_BUCKET)
     logging.debug(f"Installing all content packs for nightly flow in server {host}")
 
+    # Add deprecated packs to IGNORED_FILES list:
+    for pack_id in os.listdir(PACKS_FULL_PATH):
+        if is_pack_hidden(pack_id):
+            IGNORED_FILES.append(pack_id)
```
Comment (hod-alpert) [inline Tests/Marketplace/search_and_install_packs.py:472]:
```suggestion
            logging.debug(f'Skipping installation of hidden pack "{pack_id}"')
            IGNORED_FILES.append(pack_id)
```

**Review by hod-alpert — APPROVED** [review]

**Review by yuvalbenshalom — APPROVED** [review]

**Pull request merged**